sdk/core/Azure.Core.TestFramework/src/ClientTestFixtureAttribute.cs

```diff
@@ -96,9 +96,33 @@ private List<(TestFixtureAttribute Suite, bool IsAsync, object ServiceVersion, o
         {
             var result = new List<(TestFixtureAttribute Suite, bool IsAsync, object ServiceVersion, object Parameter)>();
 
-            if (_serviceVersions.Any())
+            SortedSet<int> relevantServiceVersions;
+            if (TestEnvironment.IsRunningInCI && RecordedTestUtilities.GetModeFromEnvironment() == RecordedTestMode.Live)
```

Comment:
This would break the pattern where mode is being passed into the RecordedTestBase ctor.  I.e. https://github.com/Azure/azure-sdk-for-net/blob/850df4dc21246f6d216e8ed472313b741a8362b7/sdk/storage/Azure.Storage.Blobs.ChangeFeed/tests/BlobChangeFeedAsyncPagableTests.cs#L21

Reply:
I'm trying to asses if it matters for pipeline runs first. If it does then we'll worry about this problem.

Reply:
Regarding this particular case. One would hardcode this to re-record but is expected to not commit that hardcoded value.
I have made that mistake few times (committed hardcoded value) and discovered it pretty late (long feedback loop). Since then I started to use `.runsettings` files to toggle between modes in VS.
Maybe we should throw a validation that if `IsRunningInCI` then hardcoding mode isn't allowed, somewhere here https://github.com/Azure/azure-sdk-for-net/blob/62f7addf55fca2ec74e4b7c5f02b128c2b811b09/sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs#L60-L69
?

Reply:
We can, but we also have automation to re-record on CI so we would need to handle that case. Most of the time PR like that would fail anyway (there would be no test environment to re-record in). The message might not be the best thought.